src/cmd/resultset.js

```diff
@@ -46,6 +46,11 @@ class ResultSet extends Command {
       //*********************************************************************************************************
       case 0xff:
         const err = packet.readError(info, this.displaySql());
+
+        //force in transaction status, since query will have created a transaction if autocommit is off
+        //goal is to avoid unnecessary COMMIT/ROLLBACK.
+        info.status |= ServerStatus.STATUS_IN_TRANS;
```

Comment:
I am not sure I understand this, why to force, and why it have created a transaction. Is this what you discussed with Marko the other day. Also it's not clear how forcing transaction status helps avoid unnecessary COMMIT/ROLLBACK.